cmake/Modules/HandleCUDAComputeCapability.cmake

```diff
@@ -8,24 +8,55 @@
 # those without the CUDA_ prefix
 
 if (NOT CUDA_TARGET_COMPUTE_CAPABILITY)
-	if("$ENV{CUDA_SM}" STREQUAL "")
-		set(ENV{CUDA_INCLUDE_DIRS} "${CUDA_INCLUDE_DIRS}")
-		set(ENV{CUDA_CUDART_LIBRARY} "${CUDA_CUDART_LIBRARY}")
-		set(ENV{CMAKE_CXX_COMPILER} "${CMAKE_CXX_COMPILER}")
-		execute_process(COMMAND bash -c "${CMAKE_CURRENT_SOURCE_DIR}/scripts/get_cuda_sm.sh"  OUTPUT_VARIABLE CUDA_TARGET_COMPUTE_CAPABILITY_) 
-	else()
-		set(CUDA_TARGET_COMPUTE_CAPABILITY_ $ENV{CUDA_SM})
+	if (CMAKE_CUDA_COMPILER_LOADED)
+		set(CUDAFILE ${CMAKE_CURRENT_BINARY_DIR}/check_cuda.cu)
+		file(WRITE ${CUDAFILE}
```

Comment:
Probably, but I am not invoking nvcc directly. Instead I make use of https://cmake.org/cmake/help/latest/command/try_run.html which tries to compile and run a file. I am not sure if it is worthwhile to go a more complex, and possible bug-prone, route just to avoid the file?

Reply:
No, it's not worth it. Let's keep it simple and use a file.